ragna/core/_rag.py

```diff
@@ -41,20 +45,69 @@ class Rag(Generic[C]):
         ```
     """
 
-    def __init__(self) -> None:
-        self._components: dict[Type[C], C] = {}
+    def __init__(
```

Comment:
Two new functionalities for the `ragna.Rag` object:

1. When creating a new chat with `Rag().chat()`, one now can also pass the display name of any loaded component instead of passing an instance or the class.
2. The object can now be created with a configuration object, which loads all configured source storages and assistants.

We had this before, but only for the API. While refactoring, it made the code a lot cleaner when moving it here.